sdk/evaluation/azure-ai-evaluation/setup.py

```diff
@@ -0,0 +1,88 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import os
+import re
+from io import open
+from typing import Any, Match, cast
+
+import pkg_resources
+from setuptools import find_packages, setup
+
+# Change the PACKAGE_NAME only to change folder and different name
+PACKAGE_NAME = "azure-ai-evaluation"
+PACKAGE_PPRINT_NAME = "Evaluation"
+
+# a-b-c => a/b/c
+PACKAGE_FOLDER_PATH = PACKAGE_NAME.replace("-", "/")
+# a-b-c => a.b.c
+NAMESPACE_NAME = PACKAGE_NAME.replace("-", ".")
+
+# Version extraction inspired from 'requests'
+with open(os.path.join(PACKAGE_FOLDER_PATH, "_version.py"), "r") as fd:
+    version = cast(Match[Any], re.search(r'^VERSION\s*=\s*[\'"]([^\'"]*)[\'"]', fd.read(), re.MULTILINE)).group(1)
+if not version:
+    raise RuntimeError("Cannot find version information")
+
+with open("README.md", encoding="utf-8") as f:
+    readme = f.read()
+with open("CHANGELOG.md", encoding="utf-8") as f:
+    changelog = f.read()
+
+setup(
+    name=PACKAGE_NAME,
+    version=version,
+    description="Microsoft Azure {} Library for Python".format(PACKAGE_PPRINT_NAME),
+    long_description_content_type="text/markdown",
+    long_description=readme + "\n\n" + changelog,
+    license="MIT License",
+    author="Microsoft Corporation",
+    author_email="azuresdkengsysadmins@microsoft.com",
+    url="https://github.com/Azure/azure-sdk-for-python",
+    keywords="azure, azure sdk",
+    classifiers=[
+        "Development Status :: 4 - Beta",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3 :: Only",
+        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
+        "License :: OSI Approved :: MIT License",
+        "Operating System :: OS Independent",
+    ],
+    zip_safe=False,
+    include_package_data=True,
+    packages=find_packages(
+        exclude=[
+            "tests",
+            # Exclude packages that will be covered by PEP420 or nspkg
+            "azure",
+            "azure.ai",
+        ]
+    ),
+    python_requires=">=3.8",
+    install_requires=[
+        # NOTE: To avoid breaking changes in a major version bump, all dependencies should pin an upper bound if possible.
+        "promptflow-devkit<2.0.0,>=1.15.0",
```

Comment:
How big is our dependency on this? You had mentioned in the review meeting that one of the reasons to extract this library from the promptflow structure is that customers don't need to use promptflow in order to use the evaluators.
This aspiration is negated if we make people install promptflow anyway. 
Can we look at whether this could be an optional extra based on specific scenarios?

Reply:
We use promptflow to do our batch execution of evaluator, so it is required for our core scenarios to work. The future plan for this promptflow dependency is being determined, so we can chat offline about it. Opened a tracking task

cc: @luigiw @singankit